api/src/ginger_calls.rs

```diff
@@ -306,22 +339,28 @@ pub fn create_naive_threshold_sig_proof(
     );
 
     //Read proving key
-    let params = read_from_file(proving_key_path)?;
+    let pk: IndexProverKey<Fr, IPAPC> = if enforce_membership {
+        read_from_file_checked(proving_key_path)
+    } else {
+        read_from_file(proving_key_path)
+    }?;
 
     //Create and return proof
     let mut rng = OsRng;
-    let proof = create_random_proof(c, &params, &mut rng)?;
+    let proof = MarlinInst::prove::<_, OsRng>(&pk, c, &mut rng).unwrap();
+
     Ok((proof, valid_signatures))
 }
 
 pub fn verify_naive_threshold_sig_proof(
     constant:                 &FieldElement,
-    end_epoch_mc_b_hash:      &[u8; 32],
-    prev_end_epoch_mc_b_hash: &[u8; 32],
+    end_epoch_mc_b_hash:      &[u8; 16],
```

Comment:
These sizes are still 32: it's a MC hash, i.e. a SHA-256, they have nothing to do with our fields

Reply:
With epoch 32 bytes length I am getting the error in UT: "Attempt to deserialize a field element over the modulus". Actually tweedle fields are 256bit long and with 256bit vector we of course can have value which is bigger then modulus of the field, The FromBytes implementation for fields in ginger-lib cannot do reduce by modulus during the reading from buffer. What should be the solution?

1. Change the implementation in ginger-lib so if modulus smaller than passed bytes, just do the reduction

2. Same thing as above, but do it in the zendoo in function "read_field_element_from_buffer_with_padding"

I prefer second variant, because first one will affect all projects based on ginger-lib in unpredictable way

Reply:
Yes you're completely right. There can be several approaches: 

1.  We truncate the byte array by 1 byte (or 2 bit) so we are able to safely deserialize a FieldElement (we lose just 4 bit/1 bit of security);
2.  We call the call the to_field_elements() function of GingerLib for [u8] (https://github.com/HorizenOfficial/ginger-lib/blob/algebra_ocl/algebra/src/to_field_vec.rs#L161), but this will deserialize 2 field elements out of this 32 byte array (and will slightly change the circuit too);
3.  Deserialize two field elements with the aforementioned function, compute the PoseidonHash and use the output as "deserialized" field element.

We need to discuss with MC and SC what is the preferred approach. As I understood they have not still reached the point in which this MC hashes are used, but in other points they need to deserialize a FieldElement from a SHA-256 hash and they are using approach number 1. 
Let's discuss it separately.